File: support/dbimport.pl
#!/usr/bin/env perl

###
## @(#) dbimport		falcon@icb.net
## Updated by hoche@grok.com       5/24/19
##
## this builds a database by importing data created by dbexport
## it's not hammered on since it's only been used once
###

dbmopen(my %DB, "./icbdb", 0666);

while (my $line = <STDIN>)
{
	if ( $line =~ /([^|]+)\|(.*)/ )
	{
		printf("%s=%s\n", $1, $2);
		$DB{$1} = $2;
	}
}

dbmclose(%DB);

File: support/dbimport.py
#!/usr/bin/env python3

###
## dbimport.py    falcon@icb.net
##
## this builds a database by importing data created by dbexport
## it's not hammered on since it's only been used once
###

import sys
import os

sys.path.insert(0, os.path.dirname(os.path.abspath(__file__)))
from icbdb import IcbDb

def main():
    dbpath = sys.argv[1] if len(sys.argv) > 1 else "./icbdb"
    with IcbDb(dbpath) as db:
        for line in sys.stdin:
            line = line.rstrip("\n")
            sep = line.find("|")
            if sep >= 0:
                key = line[:sep]
                val = line[sep + 1:]
                print(f"{key}={val}")
                db[key] = val

if __name__ == "__main__":
    main()
